k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -138,10 +156,64 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
 
     (_watch(resourceVersion), Closable.ref(close))
   }
+
+  /**
+   * Convert this Watchable into an [[Activity]]
+   * @param resourceVersion whether or not to send the initial resource
+   *                        version of the watched resource. this is a
+   *                        special case due to errors with the
+   *                        ConfigMap interpreter
```

Comment:
I think more generally you want to supply the resourceVersion when calling Watch List endpoints, but you don't want to include it when calling the Watch (object) endpoints, since it may be too old. In fact, maybe there's a way to enforce that in the code as well?